lib/src/clojure_lsp/internal_api.clj

```diff
@@ -304,8 +284,10 @@
          :message (edits->diff-string edits options db)
          :edits edits}
         (do
-          (mapv (comp #(cli-println options "Cleaned" (uri->ns (:uri %) ns+uris))
-                      #(apply-workspace-edit-summary! % components)) edits)
+          (run! #(apply-workspace-edit-summary! % components) edits)
+          (->> edits
+               (mapcat #(dep-graph/ns-names-for-uri db (:uri %)))
+               (run! #(cli-println options "Cleaned" %)))
```

Comment:
Wouldn't this change make clojure-lsp first apply all edits and only then print all applied changes? if so, maybe the user xp would be affected since we would lose the visual feedback that clojure-lsp cleaned 3/10 files and will only know that after cleaning the last 10 file, am I reading that correctly?

Reply:
I had trouble reading this too, which is why I re-wrote it. Actually, it's doing the same thing it's always done. That is, it always changed the files before printing. The thing to remember is that `comp` runs the _rightmost_ function first. So the old code first ran `apply-workspaced-edit-summary!` (which returned the edits it was passed) and then second ran `cli-println` based on those edits. But the `comp` is a weird way to do that. I also switched from `mapv` to `run!` because I think it's important to call out that these are being called for their side-effects. It's true that `mapv` forces side-effects because it has to realize a vector, but that's more obscure than `run!` or `doall`.

Reply:
You are right, I missed that `comp`. Yeah, replacing all mapvs with run! seems way better, totally agreed with that